### PR TITLE
added additional cassandra jobs to run on cron and [full build] 

### LINKED ISSUE #1594: enable cron testing of Cassandra versions covered by .travis.yml.cassandra
The Cassandra tests for version `2.2.10`, `3.0.14`, and `3.11.0` have to be manually run right now. Tests that fail in Cassandra 3 were introduced 10 months ago that I just found today. I don't believe there were any user issues as result, but it's still a good indicator that this should be run in an automated fashion. 

I propose adding the contents of `.travis.yml.cassandra` to `.travis.yml` based off the conditional that the `TRAVIS_EVENT_TYPE` environment variable has the value `cron`. I might also be able to have it run off of a commit message string similar to [skip ci]

I think running this more comprehensive test weekly would be appropriate.

### PR DESCRIPTION
**Pull request by chupman**
Signed-off-by: Chris Hupman <chupman@us.ibm.com>

fixes #1594 

Here's an [example](https://travis-ci.org/chupman/janusgraph/builds/536458452) from my branch showing how the jobs get added when run in cron.

I actually ran into issues getting it to run on tags, but I did figure out building based of commit message contents which I think is better anyways. Here's an [example](https://travis-ci.org/chupman/janusgraph/builds/536917826)

-----

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [X] Is there an issue associated with this PR? Is it referenced in the commit message?
- [X] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [X] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [X] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you written and/or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](https://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [ ] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?
- [ ] If this PR is a documentation-only change, have you added a `[skip ci]`
  tag to the first line of your commit message to avoid spending CPU cycles in
  Travis CI when no code, tests, or build configuration are modified?

### Note:
Please ensure that once the PR is submitted, you check Travis CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -34,45 +34,109 @@ env:
     # Default Elasticsearch heap size can be too large for Travis
     - ES_JAVA_OPTS="-Xms256m -Xmx512m"
 
-  matrix:
-    # sort modules by test time with quickest modules first
```
Comment (mbrukman) [inline .travis.yml]: You've lost this comment in your change; was that intentional? Would you like to bring it back?

Comment (chupman) [reply .travis.yml]: Sure. I'll add it back in.

**Comment by chupman** [comment]
This is successfully running the additional tests based off the `[full build]` string in the commit message and cron is working to trigger the additional tests as well. I put in a fix to resolve some Cassandra 3 errors in `0.2.3`. 

Once the `0.2.3` release finalizes and those fixes are merged I'll rerun travis and this should be good to go in. If any of the extended tests still fail that shouldn't prevent us from merging this as the goal of this patch is make it easier to detect those failures.

**Comment by porunov** [comment]
So, actually we don't have test failures here. Both failures are caused by:
```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-antrun-plugin:1.7:run (default-cli) on project janusgraph-hadoop-2: An Ant BuildException has occured: Unable to read execution data file /home/travis/build/JanusGraph/janusgraph/target/jacoco-combined.exec
[ERROR] around Ant part ...<report>... @ 5:11 in /home/travis/build/JanusGraph/janusgraph/janusgraph-hadoop-parent/janusgraph-hadoop-2/target/antrun/build-main.xml: /home/travis/build/JanusGraph/janusgraph/target/jacoco-combined.exec (No such file or directory)
```
Which isn't directly related by Cassandra tests. jacoco is code coverage library and looks like it is missing. I didn't use jacoco and don't know how it works but can we just either add that into JanusGraph or remove usage of jacoco completely to prevent such failures?

**Comment by porunov** [comment]
Okay there are some other errors:
```
[ERROR] testReadSelfEdge  Time elapsed: 0.001 s  <<< ERROR!
java.lang.IllegalArgumentException: Path /home/travis/build/JanusGraph/janusgraph/janusgraph-hadoop-parent/janusgraph-hadoop-2/target/cassandra/conf/localhost-bop must be a directory
	at org.janusgraph.hadoop.Cassandra3InputFormatIT.getConfiguration(Cassandra3InputFormatIT.java:36)
```
Isn't clear for me yet

**Comment by chupman** [comment]
I think jacoco is used by codecov for the badge score. 

I found a [blog post on troubleshooting jacoco](http://www.ffbit.com/blog/2014/05/21/skipping-jacoco-execution-due-to-missing-execution-data-file/) and can hopefully take a look on Monday.

**Comment by porunov** [comment]
I am thinking, maybe it is wise to enable tinkerpop tests also on a full build?
Or add another job which triggers tinkerpop tests.
It is a pain to execute them on a local machine.

https://github.com/JanusGraph/janusgraph/pull/1618#issuecomment-499961967

**Comment by FlorianHockmann** [comment]
> I am thinking, maybe it is wise to enable tinkerpop tests also on a full build?
Or add another job which triggers tinkerpop tests.

I think that's a good idea in general as it's really problematic if we merge something that breaks the TinkerPop tests and we only find out months later (like in #1466 where it was ~5 months), but the problem is that those tests take something like 8h for backends like CQL or HBase. That's why I only enabled them for the inmemory backend for the normal Travis build, but I think that we can create another travis build similar to this one that executes TinkerPop tests for more backends if we split them up into multiple jobs.
However, I would do that in a separate PR.

**Comment by FlorianHockmann** [comment]
Regarding jacoco complaining about missing test data: I had the same problem when I wanted to create a [Travis job that only executes TinkerPop tests](https://github.com/JanusGraph/janusgraph/blob/b0ba0e3102a41aa4f07238b08230e7891addc3e4/.travis.yml#L100) as that profile doesn't produce coverage data. So, I had to omit the `-Pcoverage` flag in that job, but since the job here should produce coverage data, I'm not sure what's wrong here.

**Comment by porunov** [comment]
> I think that's a good idea in general as it's really problematic if we merge something that breaks the TinkerPop tests and we only find out months later (like in #1466 where it was ~5 months), but the problem is that those tests take something like 8h for backends like CQL or HBase. That's why I only enabled them for the inmemory backend for the normal Travis build, but I think that we can create another travis build similar to this one that executes TinkerPop tests for more backends if we split them up into multiple jobs.
> However, I would do that in a separate PR.

Agree. I have created the issue #1620 to tack it

**Comment by chupman** [comment]
I think the jacoco error was a flake. I reran and was getting the following error:
```
[ERROR] testReadWideVertexWithManyProperties  Time elapsed: 0.001 s  <<< ERROR!
java.lang.IllegalArgumentException: Path /Users/chupman/github/janusgraph/janusgraph/janusgraph-hadoop-parent/janusgraph-hadoop-2/target/cassandra/conf/localhost-bop must be a directory
	at org.janusgraph.hadoop.Cassandra3InputFormatIT.getConfiguration(Cassandra3InputFormatIT.java:36)

[INFO]
[INFO] Results:
[INFO]
[ERROR] Errors:
[ERROR]   Cassandra3InputFormatIT>JanusGraphBaseTest.setUp:98->getConfiguration:36 » IllegalArgument
[ERROR]   Cassandra3InputFormatIT>JanusGraphBaseTest.setUp:98->getConfiguration:36 » IllegalArgument
[ERROR]   Cassandra3InputFormatIT>JanusGraphBaseTest.setUp:98->getConfiguration:36 » IllegalArgument
[ERROR]   Cassandra3InputFormatIT>JanusGraphBaseTest.setUp:98->getConfiguration:36 » IllegalArgument
[ERROR]   Cassandra3InputFormatIT>JanusGraphBaseTest.setUp:98->getConfiguration:36 » IllegalArgument
[INFO]
[ERROR] Tests run: 5, Failures: 0, Errors: 5, Skipped: 0
```

When I tested locally I found that `conf/localhost/bop` is getting created at `janusgraph-cassandra/target/cassandra/conf/localhost-bop`, but it's not getting copied over to `janusgraph-hadoop-parent/janusgraph-hadoop-2/target/cassandra/conf/localhost-bop`

I'm back in the office so I'll take a look. I think the fix will probably have to be in a separate PR though since I already tried to fix this by adding `INSTALL_ARGS`.

**Comment by chupman** [comment]
fixing the missing `localhost-bop` directory was incredibly straightforward, but I hit a different issue afterwards.

Here's the test output https://travis-ci.org/chupman/janusgraph/jobs/544819649#L6149-L6211.

On a hunch I also tried checking out the JanusGraph 0.3.0 tag and running against an external Cassandra as listed in [TESTING.md](https://travis-ci.org/chupman/janusgraph/jobs/544819649#L6149-L6211) on my laptop and a CentOS VM and it's failing to connect via thrift. 

```
testReadSelfEdge(org.janusgraph.hadoop.Cassandra3InputFormatIT)  Time elapsed: 5.677 sec  <<< ERROR!
java.lang.IllegalStateException: org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 0.0 failed 1 times, most recent failure: Lost task 0.0 in stage 0.0 (TID 0, localhost, executor driver): java.lang.RuntimeException: error communicating via Thrift
```

I don't think these 2 tests have ever been run against the 0.3.x branch and appear to have been broken since 0.3.0. I'm not sure if we want to block the releases because of this of if I should just comment out the 2 failing tests and open an issue to fix them. 

@sjudeng do you have insight on how to fix these tests perchance?

**Comment by chupman** [comment]
@FlorianHockmann @porunov the 2 broken tests are now showing up correctly in the allowed failures section. Once the tests finish this should be good to go.

After the releases are done I recommend we add weekly cron jobs for 0.3 and master to run on Saturdays since CI tends to be light on the weekend.

**Comment by porunov** [comment]
@chupman Thank you. Could you please create issues to reference these failed tests so that we could fix them later?

**Pull request merged**